### PR TITLE
Apollo 2.0.5 + default_group + fixes

### PR DESCRIPTION
**Pull request by abretaud**
Hellon
Several changes in this PR:
- added support for tokens to make it work with Apollo 2.0.5
- now it's possible to grant permissions to a group while createing/updating an organism
- working with conda resolver now (still need python modules in .venv for dynamic select boxes)
As usual, comments welcome :)

### CODE REVIEW

**Comment by nathandunn** [comment]
@abretaud  This looks great.  Just a quick note.  In 2.0.5 we had this bug (which you seem to have been able to work around) that required an organism to fix:

https://github.com/GMOD/Apollo/issues/1388 

Also, curious about how you made the bioconda integration work. 

I'll let @erasche do the merging.

File: create_or_update_organism.xml [hunk 1]
```diff
@@ -29,7 +33,16 @@ $__user_email__
...
+    <conditional name="perms">
+        <param name="public" type="select" label="Is Organism Public">
+            <option value="no" selected="true">No</option>
+            <option value="yes">Yes</option>
+        </param>
+        <when value="yes"/>
+        <when value="no">
+            <param name="group" type="text" label="Grant access to a user group" optional="True" />
+        </when>
+    </conditional>
```
Comment (erasche) [inline create_or_update_organism.xml]: Should this be a mutually exclusive option? Perhaps there is a use case where people want to share edit access with a group and simultaneously make it a public organism?

Additionally, would you have time / be willing to make the group select a code_file function as well? E.g.

```xml
<param name="group_select" type="select" dynamic_options="galaxy_list_groups(__trans__)" label="Group" />
```

and an associated function in webapollo.py. If not, I can find time to add this as well.

File: webapollo.py [hunk 2]
```diff
@@ -162,6 +164,7 @@ def request(self, clientMethod, data, post_params={}, isJson=True):
         data.update({
             'username': self._wa.username,
             'password': self._wa.password,
+            'clientToken': self._wa.clientToken,
```
Comment (erasche) [inline webapollo.py]: interesting!

File: create_or_update_organism.xml (hunk 1 above)
Comment (abretaud) [inline create_or_update_organism.xml]: Yep, writing the dynamic option is a good idea, I'll try to do it.

For the options being exclusive or not, to be honest I haven't work enough with apollo 2 yet to decide, @nathandunn what do you think?

**Comment by abretaud** [comment]
@nathandunn ok for the bug, so if I understood well, in the next release we could just remove the token stuff I added in this PR?

For the conda stuff, I have a (temporary) branch of docker-galaxy-annotation there if you want to test it: https://github.com/abretaud/docker-galaxy-annotation/tree/bipaa  (image there: https://quay.io/repository/abretaud/galaxy-genome-annotation)

File: webapollo.py (hunk 2 above)
Comment (nathandunn) [inline webapollo.py]: right, this is part of the 2.0.5 bug (I think).  Should be fixed in 2.0.6 coming out soon.

**Comment by erasche** [comment]
Can't really upgrade to a pre-release, so I'm merging as-is and will add the group enumeration code myself. Thanks for the PR @abretaud!

**Comment by erasche** [comment]
Made changes to include the group selection. Additionally made the option non-exclusive since this seems more widely useful.

File: create_or_update_organism.xml (hunk 1 above)
Comment (nathandunn) [inline create_or_update_organism.xml]: @erasche  / @abretaud  Sorry for the late reply (just noticed this comment).  These are slightly orthogonal permissions.   If an organism is public the world can see the evidence for that organism (similar to JBrowse), but not the annotations.   

When you grant a group permission, at a minimum they'll be able to see the annotations + evidence.

Comment (erasche) [inline create_or_update_organism.xml]: Oh? I always assumed public = public annotations as well. Is there not a permission that permits that? People generally don't care about our evidence tracks, they're there for the annotations themselves.

**Comment by nathandunn** [comment]
Funny.  This came up just the other day in @deepakunni3 's lab .  

The way to do this (and maybe what you were doing) is to create a group and assign all of the user's  (or a "demo user") to that group and give them read-only permissions on the different organisms you are interested in. 

If you wanted to open an issue to emulate this that would be fine.  I'm not sure when I would get to it.

**Pull request merged**